it/uk/gov/hmrc/agentrelationships/controllers/RelationshipsControllerISpec.scala

```diff
@@ -483,124 +479,125 @@ class RelationshipISpec extends UnitSpec
     }
 
-    "return 404 when agent code is not found in gg" in {
-      givenAgentCredentialsAreFoundFor(Arn(arn), "foo")
-      givenAgentCodeIsNotInTheResponseFor("foo")
-      givenAgentIsAllocatedAndAssignedToClient(vrn, "bar")
-      givenAuditConnector()
+    "return 404 when agent code is not found in ugs" in {
+      givenPrincipalUser(arn, "foo")
+      givenGroupInfoNotExists("foo")
+      givenDelegatedGroupIdsExistFor(vrn, Set("foo"))
+      givenDelegatedGroupIdsNotExistForKey(s"HMCE-VATDEC-ORG~VATRegNo~${vrn.value}")
+
       val result = await(doRequest)
       result.status shouldBe 404
-      (result.json \ "code").as[String] shouldBe "UNKNOWN_AGENT_CODE"
+      (result.json \ "code").as[String] shouldBe "RELATIONSHIP_NOT_FOUND"
     }
 
     //FAILURE CASES
 
-    "return 502 when GsoAdminGetCredentialsForDirectEnrolments returns 5xx" in {
-      whenGetCredentialsReturns(500)
-      givenAgentCodeIsFoundFor("foo", "bar")
+    "return 502 when ES1/principal returns 5xx" in {
+      givenPrincipalGroupIdRequestFailsWith(500)
+      givenGroupInfo("foo", "bar")
       givenAgentIsAllocatedAndAssignedToClient(vrn, "bar")
-      givenAuditConnector()
+
       val result = await(doRequest)
       result.status shouldBe 502
     }
 
-    "return 502 when GsoAdminGetUserDetails returns 5xx" in {
-      givenAgentCredentialsAreFoundFor(Arn(arn), "foo")
-      whenGetUserDetailReturns(500)
+    "return 502 when UGS returns 5xx" in {
+      givenPrincipalUser(arn, "foo")
+      givenGroupInfoFailsWith(500)
       givenAgentIsAllocatedAndAssignedToClient(vrn, "bar")
-      givenAuditConnector()
+
       val result = await(doRequest)
       result.status shouldBe 502
     }
 
-    "return 502 when GsoAdminGetAssignedAgents returns 5xx" in {
-      givenAgentCredentialsAreFoundFor(Arn(arn), "foo")
-      givenAgentCodeIsFoundFor("foo", "bar")
-      givenAuditConnector()
-      whenGetAssignedAgentsReturns(500)
+    "return 502 when ES1/delegated returns 5xx" in {
+      givenPrincipalUser(arn, "foo")
+      givenGroupInfo("foo", "bar")
+
+      givenDelegatedGroupIdRequestFailsWith(500)
       val result = await(doRequest)
       result.status shouldBe 502
     }
 
-    "return 400 when GsoAdminGetCredentialsForDirectEnrolments returns 4xx" in {
-      whenGetCredentialsReturns(400)
-      givenAgentCodeIsFoundFor("foo", "bar")
+    "return 400 when ES1/principal returns 4xx" in {
+      givenPrincipalGroupIdRequestFailsWith(400)
+      givenGroupInfo("foo", "bar")
       givenAgentIsAllocatedAndAssignedToClient(vrn, "bar")
-      givenAuditConnector()
+
       val result = await(doRequest)
       result.status shouldBe 400
     }
 
-    "return 400 when GsoAdminGetUserDetails returns 4xx" in {
-      givenAgentCredentialsAreFoundFor(Arn(arn), "foo")
-      whenGetUserDetailReturns(400)
+    "return 400 when UGS returns 4xx" in {
+      givenPrincipalUser(arn, "foo")
+      givenGroupInfoFailsWith(400)
```

Comment:
`givenGroupInfoFailsWith(400)` potentially conflicts with the
`givenGroupInfo("foo", "bar")` stub called in `givenAgentIsAllocatedAndAssignedToClient(_, "bar")`